operator/controllers/http/httpscaledobject_controller_test.go

```diff
@@ -0,0 +1,73 @@
+package http
+
+import (
+	"github.com/kedacore/http-add-on/operator/controllers/http/config"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestHttpScaledObjectController(t *testing.T) {
+	r := require.New(t)
+
+	testInfra := newCommonTestInfra("testns", "testapp")
+
+	reconciller := &HTTPScaledObjectReconciler{
+		Client:               testInfra.cl,
+		Scheme:               testInfra.cl.Scheme(),
+		ExternalScalerConfig: config.ExternalScaler{},
+		BaseConfig:           config.Base{},
+	}
+
+	// Create required app objects for the application defined by the CRD
+	err := reconciller.createOrUpdateApplicationResources(
+		testInfra.ctx,
+		testInfra.logger,
+		testInfra.cl,
+		config.Base{},
+		config.ExternalScaler{},
+		&testInfra.httpso,
+	)
+	r.NoError(err)
+
+	// check for scaledobject, expect no error as scaledobject should get created
+	_, err = getSO(
+		testInfra.ctx,
+		testInfra.cl,
+		testInfra.httpso,
+	)
+	r.NoError(err)
+
+}
+
+func TestHttpScaledObjectControllerWhenSkipAnnotationSet(t *testing.T) {
```

Comment:
Could we add a test covering the scenario where the ScaledObject is removed and a normal scenario where a regular ScaledObject (not deleted). Adding those 2 scenarios we can cover that everything works as expected

Reply:
sure, will take a look at this

Reply:
@JorTurFer just looking at this, would you mind on expanding/clarifying on what tests you require for this as I'm not 100% on what your'e asking?   The 2 tests I created are already checking if the scaledObject exists or not depending if annotation is set

Reply:
sorry, I didn't see the test above (`TestHttpScaledObjectController`) which is one of them indeed. The other test that I said is:
- Deploy a regular HTTPScaledObject
- Check that the SO is correctly generated
- Update the HTTPScaledObject with the skip flag
- Check that the SO has been removed

Reply:
added a new test for above if you can review, thanks